pulsar-client-cpp/include/pulsar/Auth.h

```diff
@@ -43,33 +61,49 @@ namespace pulsar {
     protected:
         AuthenticationDataProvider();
     };
-    
+
     typedef boost::shared_ptr<AuthenticationDataProvider> AuthenticationDataPtr;
-    
+
     class Authentication {
     public:
         virtual ~Authentication();
         virtual const std::string getAuthMethodName() const = 0;
-        virtual Result getAuthData(AuthenticationDataPtr& authDataContent) const = 0;
-        
+        virtual Result getAuthData(AuthenticationDataPtr& authDataContent) const {
+            authDataContent = authData_;
+            return ResultOk;
+        }
+        // for compatibility with previous version
+        virtual Result getAuthData(std::string& authDataContent) const {
+            return ResultOk;
+        }
     protected:
         Authentication();
+        AuthenticationDataPtr authData_;
     };
-    
+
     typedef boost::shared_ptr<Authentication> AuthenticationPtr;
-    
+    typedef std::map<std::string, std::string> ParamMap;
+
     class Auth {
     public:
-        static AuthenticationPtr Disabled();
+        static AuthenticationPtr getDisabled();
         static AuthenticationPtr create(const std::string& dynamicLibPath);
         static AuthenticationPtr create(const std::string& dynamicLibPath, const std::string& authParamsString);
         static AuthenticationPtr create(const std::string& dynamicLibPath, ParamMap& params);
+        // for compatibility with previous version
+        static AuthDataPtr Disabled();
+
+// for compatibility with previous version
+#ifndef PULSAR_DISABLE_YCA
```

Comment:
YCA macro logic should be reversed, like `PULSAR_ENABLE_YCA` with default false.